src/model/test/ZoneHVACTerminalUnitVariableRefrigerantFlow_GTest.cpp

```diff
@@ -126,3 +134,160 @@ TEST_F(ModelFixture, ZoneHVACTerminalUnitVariableRefrigerantFlow_SupplementalHea
   EXPECT_FALSE(vrf.setMaximumOutdoorDryBulbTemperatureforSupplementalHeaterOperation(45.0));  // > max
   EXPECT_EQ(19.0, vrf.maximumOutdoorDryBulbTemperatureforSupplementalHeaterOperation());
 }
+
+TEST_F(ModelFixture, ZoneHVACTerminalUnitVariableRefrigerantFlow_addToNode_MainBranch) {
+  Model m;
+  ZoneHVACTerminalUnitVariableRefrigerantFlow testObject(m);
+
+  AirLoopHVAC airLoop(m);
+
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+
+  EXPECT_TRUE(testObject.addToNode(supplyOutletNode));
+
+  EXPECT_EQ((unsigned)3, airLoop.supplyComponents().size());
+
+  EXPECT_TRUE(testObject.inletPort());
+  EXPECT_TRUE(testObject.outletPort());
+  EXPECT_TRUE(testObject.airLoopHVAC());
+}
+
+TEST_F(ModelFixture, ZoneHVACTerminalUnitVariableRefrigerantFlow_AddToNodeTwoSameObjects) {
+  Model m;
+  ZoneHVACTerminalUnitVariableRefrigerantFlow testObject(m);
+
+  AirLoopHVAC airLoop(m);
+
+  Node supplyOutletNode = airLoop.supplyOutletNode();
+  testObject.addToNode(supplyOutletNode);
+  supplyOutletNode = airLoop.supplyOutletNode();
+  EXPECT_FALSE(testObject.addToNode(supplyOutletNode));
+
+  EXPECT_TRUE(testObject.inletPort());
+  EXPECT_TRUE(testObject.outletPort());
+}
+
+TEST_F(ModelFixture, ZoneHVACTerminalUnitVariableRefrigerantFlow_AddToNodeAirLoopDemandSide) {
+  Model m;
+  ZoneHVACTerminalUnitVariableRefrigerantFlow testObject(m);
+
+  AirLoopHVAC airLoop(m);
+
+  Node inletNode = airLoop.zoneSplitter().lastOutletModelObject()->cast<Node>();
+
+  EXPECT_FALSE(testObject.addToNode(inletNode));
```

Comment:
Check that we correctly disallow connecting on the demand side of an AirLoopHVAC